File: youlexuan-manager-web/src/main/java/com/offcn/sellergoods/controller/BrandController.java
package com.offcn.sellergoods.controller;

import com.alibaba.dubbo.config.annotation.Reference;
import com.offcn.entity.PageResult;
import com.offcn.entity.Result;
import com.offcn.pojo.TbBrand;
import com.offcn.sellergoods.service.BrandService;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;

/*
品牌控制层
 */
@RestController
@RequestMapping("/brand")
public class BrandController {

    @Reference
    private BrandService brandService;

    /*
    全部品牌列表
     */
    @RequestMapping("/selectAll")
    public List<TbBrand> selectAll() {
        return brandService.selectAll();
    }

    /*
      分页列表
     */
    @RequestMapping("/selectPage")
    public PageResult selectPage(int page, int rows) {
        return brandService.selectPage(page, rows);
    }

    /*
   增加品牌
    */
    @RequestMapping("/insert")
    public Result insert(@RequestBody TbBrand brand){
        try {
            brandService.insert(brand);
            return new Result(true,"增加成功");
        } catch (Exception e) {
            e.printStackTrace();
            return new Result(false,"增加失败");
        }
    }

    /*
    修改品牌
     */
    @RequestMapping("/update")
    public Result update(@RequestBody TbBrand brand){
        try {
            brandService.update(brand);
            return new Result(true,"修改成功");
        } catch (Exception e) {
            e.printStackTrace();
            return new Result(false,"修改失败");
        }
    }

    /*
    根据ID获取实体
     */
    @RequestMapping("/selectOne")
    public TbBrand selectOne(Long id){
        return brandService.selectOne(id);
    }

     /*
   删除品牌
    */
     @RequestMapping("/delete")
     public Result delete(Long[] ids){
         try {
             brandService.delete(ids);
             return new Result(true,"删除成功");
         } catch (Exception e) {
             e.printStackTrace();
             return new Result(true,"删除失败");
         }
     }

     /*
    条件查询
    */
     @RequestMapping("/search")
     public PageResult search(@RequestBody TbBrand brand, int page, int rows  ){
         return brandService.findPage(brand, page, rows);
     }

}

File: youlexuan-manager-web/src/main/webapp/js/controller/brandController.js
/*
定义controller
 */
app.controller('brandController',function ($scope,$http,brandService,$controller) {

    //声明:brandController需要继承baseControler中的组件
    $controller('baseController',{$scope:$scope});

    //获取所有的品牌信息
    $scope.selectAll=function () {
        brandService.selectAll().success(
            function (response) {
                $scope.brandList=response;
            }
        )
    };

    //分页
    $scope.selectPage=function (page,rows) {
        brandService.selectPage(page,rows).success(
            function (response) {
                $scope.brandList=response.rows;
                $scope.paginationConf.totalItems=response.total;
            }
        )
    };


    //增加/更新品牌
    $scope.save=function () {
        if($scope.entity.name=='' || $scope.entity.name == null){
            alert("品牌名称不能为空");
            return;
        }
        if($scope.entity.firstChar=='' || $scope.entity.firstChar == null){
            alert("首字母不能为空");
            return;
        }
        if($scope.entity.id!=null){
            brandService.update($scope.entity).success(
                function (response) {
                    if(response.success){
                        $scope.entity={name:'',firstChar:''};
                        $scope.reloadList();
                    }else{
                        alert(response.message);
                    }
                }
            )
        }else{
            brandService.insert($scope.entity).success(
                function (response) {
                    if(response.success){
                        $scope.entity={name:'',firstChar:''};
                        $scope.reloadList();
                    }else{
                        alert(response.message);
                    }
                }
            )
        }
    };

    //查询单个
    $scope.selectOne=function (id) {
        brandService.selectOne(id).success(
        function (response) {
            $scope.entity=response;
        }
    )
};

    //删除品牌
    $scope.delete=function () {
        if($scope.selectIds==null||$scope.selectIds.length==0){
            alert("请选择后进行删除");
            return;
        }
        brandService.delete($scope.selectIds).success(
            function (response) {
                if(response.success){
                    $scope.reloadList();
                    $scope.selectIds=[];
                }else{
                    alert(response.message);
                }
            }
        )
    };

    //条件搜索
    $scope.searchEntity={};
    $scope.search=function(page,rows) {
        brandService.search(page,rows,$scope.searchEntity).success(
            function (response) {
                $scope.paginationConf.totalItems = response.total;
                $scope.brandList = response.rows;
            }
        );
    };

})